### PR TITLE
Project 3: Add and remove

### PR DESCRIPTION
**Pull request by crgc**
### Features

- Removed all hardcoded items from the tasks array.
- Create a new JS file for the new functionality: `crud.js`
- Implemented a function for adding a new task
- Implemented a function for deleting a task
- Implemented a function for the "Clear all completed" button using the `filter()` method
- By default, new tasks have `completed` set to `false` and `index` set to the value of the new array length
- When a task is deleted, the remaining indices are updated to reflect the new length
- All changes are saved using `localStorage`

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -58,10 +60,65 @@ function buildTaskLeftElement(task) {
...
+
+    const taskIndex = targetElement.id.substring('edit-task-'.length);
+    const deleteElement = document.getElementById(`delete-task-${taskIndex}`);
+
+    taskElement.classList.add('editable');
+    targetElement.classList.add('hidden');
+    deleteElement.classList.remove('hidden');
+
+    const tasks = loadTasks();
+    const editableTask = tasks.filter((task) => task.index == taskIndex )[0]; /* eslint-disable-line  */
```
Comment (voscarmv) [inline src/index.js]: - [x] Please avoid unnecessary linter disabling. Here, for example, you can use `===` by converting the data type of the variables. Please read this article on how strict comparisons and variable types work, to figure out how to do this: https://learn.onemonth.com/difference-between-equal-signs-javascript/

File: src/index.js [hunk 2]
```diff
@@ -58,10 +60,65 @@ function buildTaskLeftElement(task) {
...
+  });
+
+  const deleteElement = createElementWithClass('i', 'fa fa-trash hidden');
+  deleteElement.setAttribute('id', `delete-task-${task.index}`);
+  deleteElement.setAttribute('aria-hidden', 'true');
+  deleteElement.addEventListener('click', (e) => {
+    const taskIndex = e.target.id.substring('delete-task-'.length);
+
+    let tasks = loadTasks();
+    tasks = tasks.filter((task) => task.index != taskIndex); /* eslint-disable-line eqeqeq */
```
Comment (voscarmv) [inline src/index.js]: - [x] Please avoid unnecessary linter disabling. Here, for example, you can use `!==` by converting the data type of the variables. Please read this article on how strict comparisons and variable types work, to figure out how to do this: https://learn.onemonth.com/difference-between-equal-signs-javascript/

File: src/index.js [hunk 3]
```diff
@@ -58,10 +60,65 @@ function buildTaskLeftElement(task) {
...
+
+    let shiftedTasks = [];
+    for (let i = 0; i < tasks.length; i++) { /* eslint-disable-line no-plusplus */
```
Comment (voscarmv) [inline src/index.js]: - [x] Please avoid unnecessary linter disabling. Please use `i += 1` instead.

**Review by voscarmv — CHANGES_REQUESTED** [review]
Hi @crgc , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- Please avoid unnecessary linter disabling.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Adding a Live Demo link to your README.
- Allowing the `package.json` file to your repo in order to avoid instructions about installing all the necessary NPM packages manually. (with package.json you can just run npm install and that will install all the necessary dependencies) Read this for details https://nodejs.dev/learn/the-package-json-guide

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (crgc) [reply src/index.js]: Done.

File: src/index.js (hunk 2 above)
Comment (crgc) [reply src/index.js]: Done.

File: src/index.js (hunk 3 above)
Comment (crgc) [reply src/index.js]: Done.

**Review by mricanho — APPROVED** [review]
# Status: ✔️ Project Approved ✔️ 

Hello @crgc  🔥 

You worked hard in all the instructions that the previous Code Reviewers give you👍🏽 

Now you can merge this branch 🚀 

### 💻 👩🏽‍💻  Happy Coding!  👨🏽‍💻 🖥️

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**